### PR TITLE
Add clang-format file and update code style

### PR DESCRIPTION
**Pull request by peterbell10**
Confession time. The only reason that `uarray`'s C++ code has such a weird indent style is that I had my editor setup to work on [`pypocketfft`](https://gitlab.mpcdf.mpg.de/mtr/pypocketfft/blob/master/pypocketfft.cc#L50) when I was writing it...  This should hopefully be a bit less weird.

I could even make this a check in CI if you think that makes sense. It would certainly match with the `pytest-black` stuff in the python code.

### CODE REVIEW

**Comment by hameerabbasi** [comment]
I think a separate CI entry that checked C++ formatting would be nice. 😄

**Comment by peterbell10** [comment]
Yes, I haven't forgotten about this; I've just been busy working on a report that's due today. I should be able to add the CI check either this evening or at the weekend.

File: ci/azure-run-clang-format.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+steps:
+- bash: |
+    wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
+    sudo add-apt-repository "deb http://apt.llvm.org/xenial/ llvm-toolchain-xenial-8 main"
+    sudo apt-get update
+    sudo apt-get install -y clang-format-8
+  displayName: Install clang-format
+- bash: |
+    clang-format-8 -i uarray/_uarray_dispatch.cxx
+    if [ "$(git diff)" != "" ]; then exit 1; fi
```
Comment (hameerabbasi) [inline ci/azure-run-clang-format.yml]: It'd be nice to print the diff to see what's wrong.

**Comment by peterbell10** [comment]
The CI was supposed to fail on the latest commit, this is broken.  I was trying to push a fix but couln't use git because my syslog was apparently blowing up to 370 GB and I ran out of disk space.

**Pull request merged**